cbdr/cortex
# EM-1185: Webpage Index Issues

## Purpose:
This does two things:
- Disables indexing of the `Webpages` `tables_widget` field. ElasticSearch was breaking when attempting to index records with data in this column, as ES dynamically inferred that it was a nested datatype, and was attempting to analyze and index its nested structure. Because each Webpage has potentially vastly different `tables_widget` data structures, ES's initially inferred nested structure (which it would infer from the first record it came across with `tables_widget` data) was invalid for subsequent records. There are native ways in ES to set up dynamic mappings of these nested structures, but we do not need `tables_widget` to be indexed. It's worth looking into their dynamic mapping system when we implement ES for Cortex Beta, however.
- As a bonus, explicitly defines mappings for all `Webpages` fields, and disables tokenization/analysis of more fields

## JIRA:
https://cb-content-enablement.atlassian.net/browse/EM-1185

## Steps to Take On Prod
This has already been tested on Production (as it was the only place it was happening), is in place and works.

## Changes:
* Changes to setup
  * N/A

* Architectural changes
  * N/A

* Migrations
  * N/A

* Library changes
  * N/A

* Side effects
  * N/A

## Screenshots
* Before
N/A

* After
N/A

## QA Links:
Since this is already on Prod and works, I will not be pushing this to a feature server. I have, however, provided verification steps below if you're interested in testing locally.

## How to Verify These Changes
* Specific pages to visit
  * N/A

* Steps to take
  * Check out code locally
  * Create two example `Webpages` with *different* `tables_widget` data
  * Reindex, ensure there are no errors
  * Ensure Webpages can be searched and updated

* Responsive considerations
  * N/A

## Relevant PRs/Dependencies:
N/A

## Additional Information
N/A
